tests/test_diag.py

```diff
@@ -64,95 +96,61 @@ def test_get_store_file(uri, expected):
     assert result == expected
 
 
-@mock.patch("unified_graphics.diag.S3Map", autospec=True)
-@mock.patch("unified_graphics.diag.S3FileSystem", autospec=True)
-def test_get_store_s3(mock_s3filesystem, mock_s3map):
-    prev_env = dict(**os.environ)
-    key = "test-key"
-    token = "test-token"
-    secret = "test-secret"
+def test_get_store_s3(moto_server, s3_client, monkeypatch):
     client = {"region_name": "us-east-1"}
     uri = "s3://bucket/prefix/diag.zarr"
+    s3_client.create_bucket(Bucket="bucket")
+    s3_client.put_object(Bucket="bucket", Body=b"Test object", Key="prefix/diag.zarr")
 
-    os.environ["AWS_ACCESS_KEY_ID"] = key
-    os.environ["AWS_SECRET_ACCESS_KEY"] = secret
-    os.environ["AWS_SESSION_TOKEN"] = token
+    monkeypatch.setattr(
+        diag,
+        "S3FileSystem",
+        partial(diag.S3FileSystem, endpoint_url=moto_server),
+    )
```

Comment:
That's a neat way to update the S3FileSystem's `endpoint_url` in the code under test.

Reply:
Thanks. I actually think this might be a code smell, though. 😅 Ideally, we’d somehow be passing in information about the store so that in tests we could just pass this in as an argument rather than having to monkeypatch someone else’s code.